src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -909,6 +912,72 @@ private IEnumerable<CompletionItem> GetObjectBodyCompletions(TypeSymbol typeSymb
             }
         }
 
+        private IEnumerable<CompletionItem> GetDisableNextLineDiagnosticsDirectiveCompletion(BicepCompletionContext context)
+        {
+            if (context.Kind.HasFlag(BicepCompletionContextKind.DisableNextLineDiagnosticsDirectiveStart))
+            {
+                yield return CreateKeywordCompletion(LanguageConstants.DisableNextLineDiagnosticsKeyword, "Disable next line diagnostics directive", context.ReplacementRange);
+            }
+        }
+
+        private IEnumerable<CompletionItem> GetDisableNextLineDiagnosticsDirectiveCodesCompletion(SemanticModel model, BicepCompletionContext context)
+        {
+            if (context.Kind.HasFlag(BicepCompletionContextKind.DisableNextLineDiagnosticsCodes))
+            {
+                foreach (var diagnostic in GetDiagnosticCodes(context.ReplacementRange, model))
+                {
+                    yield return CreateKeywordCompletion(diagnostic.Code, diagnostic.Message, context.ReplacementRange);
+                }
+            }
+        }
+
+        private IEnumerable<IDiagnostic> GetDiagnosticCodes(Range range, SemanticModel model)
+        {
+            var lineStarts = model.SourceFile.LineStarts;
+            var position = GetPosition(range, lineStarts);
+            (var line, _) = TextCoordinateConverter.GetPosition(lineStarts, position);
+            var nextLineSpan = GetNextLineSpan(lineStarts, line, model.SourceFile.ProgramSyntax);
+
+            if (nextLineSpan is null)
+            {
+                return Enumerable.Empty<IDiagnostic>();
+            }
+
+            return model.GetAllDiagnostics()
+                .Where(diagnostic => nextLineSpan.ContainsInclusive(diagnostic.Span.Position) && nextLineSpan.ContainsInclusive(diagnostic.Span.GetEndPosition()))
```

Comment:
Shouldn't we just need to ensure the start of the diagnostic is on the next line?

I tried the following and the suppression works, but the completion doesn't - would be good to add a test case for it:
```bicep
#disable-next-line prefer-interpolation
var foo = concat('abc'/*
  */, 'asd')
```

Reply:
Yes, that's correct. Updated here:
https://github.com/Azure/bicep/commit/bbbfe09c6fa09e5f5f80af7fcaf9b487d92a7cd3

Thanks!

Reply:
@anthony-c-martin , I made changes to support "  #|" as part of below commit:
https://github.com/Azure/bicep/commit/69b0fdc1d9c12b54131fa0a1a1a8e233f383bc14

Could you please take a look? Thanks!